patches/kaggle.py

```diff
@@ -0,0 +1,40 @@
+import os
+from google.auth import credentials
+from google.cloud import bigquery
+from google.cloud.bigquery._http import Connection
+
+
+class KaggleKernelCredentials(credentials.Credentials):
+        """Custom Credentials used to authenticate using the Kernel's connected OAuth account."""
```

Comment:
Include sample usage.  Same for other public classes below.

While at it:  Could you update the PR title/description to match the new approach?